### PR TITLE
Docs: Fixed list index order and removed heroku info

### LINKED ISSUE #2284: Docs: Remove the "Deploy to Heroku" button from the Quickstart guide
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

Since Heroku removed their free tier and the button + documentation need updating, we should remove the deploy button from the [Quickstart page](https://docs.medusajs.com/quickstart/quick-start).

### How can this issue be resolved?

1. Remove the deploy button with the text above it "Alternatively,..." in the [Quickstart documentation](https://docs.medusajs.com/quickstart/quick-start).

### Are you interested in working on this issue?

- [ ] I would like to fix this issue

### PR DESCRIPTION
**Pull request by alesbe**
This PR fixes #2284 and #2306

### CODE REVIEW

File: docs/content/starters/nextjs-medusa-starter.md [hunk 1]
```diff
@@ -22,13 +22,13 @@ This document assumes you already have a Medusa server installed. If you don’t
 npx create-next-app -e https://github.com/medusajs/nextjs-starter-medusa my-medusa-storefront
 ```
 
-1. Change to the newly created directory `my-medusa-storefront` and rename the template environment variable file to use environment variables in development:
+2. Change to the newly created directory `my-medusa-storefront` and rename the template environment variable file to use environment variables in development:
```
Comment (shahednasser) [inline docs/content/starters/nextjs-medusa-starter.md:25]:
```suggestion
2\. Change to the newly created directory `my-medusa-storefront` and rename the template environment variable file to use environment variables in development:
```

File: docs/content/starters/nextjs-medusa-starter.md [hunk 2]
```diff
@@ -22,13 +22,13 @@ This document assumes you already have a Medusa server installed. If you don’t
...
 
 ```bash
 mv .env.template .env.local
 ```
 
-1. Make sure the Medusa server is running, then run the local Next.js server:
+3. Make sure the Medusa server is running, then run the local Next.js server:
```
Comment (shahednasser) [inline docs/content/starters/nextjs-medusa-starter.md:31]:
```suggestion
3\. Make sure the Medusa server is running, then run the local Next.js server:
```

**Comment by shahednasser** [comment]
@alesbe thank you for your work but can you open this PR against the `master` branch? This is mentioned in our [docs contribution guidelines](https://docs.medusajs.com/contribution-guidelines#base-branch).

**Review by shahednasser — CHANGES_REQUESTED** [review]

**Comment by shahednasser** [comment]
@alesbe can you change the base branch to `master`? You should be able to do it by clicking `edit` at the top of the PR. You can alternatively open another PR on the `master` branch and close this one.

**Comment by alesbe** [comment]
Sure! I made the change already

**Comment by shahednasser** [comment]
It seems so many changes will go into it. I'll close this PR. Can you please re-fork the repo, make the same changes you made, then create a new PR against the `master` branch?

**Pull request closed without merging**